cmd/util/cmd/epochs/cmd/recover.go

```diff
@@ -0,0 +1,186 @@
+	config, err := common.NewFlowClientConfig(flagAnAddress, flagAnPubkey, flow.ZeroID, false)
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to create flow client config")
+	}
+
+	flowClient, err := common.FlowClient(config)
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to create flow client")
+	}
+
+	snapshot, err := common.GetSnapshot(context.Background(), flowClient)
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed")
+	}
+
+	// extract arguments from recover epoch tx from snapshot
+	txArgs := extractRecoverEpochArgs(snapshot)
+
+	// encode to JSON
+	encodedTxArgs, err := epochcmdutil.EncodeArgs(txArgs)
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not encode recover epoch transaction arguments")
+	}
+
+	// write JSON args to stdout
+	_, err = stdout.Write(encodedTxArgs)
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not write jsoncdc encoded arguments")
+	}
+}
+
+// extractResetEpochArgs extracts the required transaction arguments for the `resetEpoch` transaction
+func extractRecoverEpochArgs(snapshot *inmem.Snapshot) []cadence.Value {
+	epoch := snapshot.Epochs().Current()
+	ids, err := epoch.InitialIdentities()
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to get initial identities for current epoch")
+	}
+
+	currentEpochDKG, err := epoch.DKG()
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to get DKG for current epoch")
+	}
+
+	log.Info().Msg("collecting partner network and staking keys")
+	partnerNodes := common.ReadPartnerNodeInfos(log, flagPartnerWeights, flagPartnerNodeInfoDir)
+	log.Info().Msg("")
+
+	log.Info().Msg("generating internal private networking and staking keys")
+	internalNodes := common.ReadInternalNodeInfos(log, flagInternalNodePrivInfoDir, flagConfig)
+	log.Info().Msg("")
+
+	log.Info().Msg("computing collection node clusters")
+	_, clusters, err := common.ConstructClusterAssignment(log, partnerNodes, internalNodes, flagCollectionClusters)
+	if err != nil {
+		log.Fatal().Err(err).Msg("unable to generate cluster assignment")
+	}
+	log.Info().Msg("")
+
+	epochCounter, err := epoch.Counter()
+	if err != nil {
+		log.Fatal().Err(err).Msg("unable to get epoch counter from current epoch")
+	}
+	log.Info().Msg("constructing root blocks for collection node clusters")
+	clusterBlocks := run.GenerateRootClusterBlocks(epochCounter, clusters)
+	log.Info().Msg("")
+
+	log.Info().Msg("constructing root QCs for collection node clusters")
+	clusterQCs := common.ConstructRootQCsForClusters(log, clusters, internalNodes, clusterBlocks)
+	fmt.Sprintf("", clusterQCs)
+	log.Info().Msg("")
+
+	randomSource, err := epoch.RandomSource()
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to get random source for current epoch")
+	}
+	randomSourceCdc, err := cadence.NewString(string(randomSource))
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to get random source cadence string")
+	}
+
+	dkgPubKeys := make([]cadence.Value, 0)
```

Comment:
The first element here will need to be the group public key (see https://github.com/onflow/flow-go/blob/master/model/convert/service_event.go#L360-L362).

Reply:
https://github.com/onflow/flow-go/pull/5576/commits/82c5803847db73bf462674ce37ff80765dfab159